src/nupic/regions/PyRegion.cpp

```diff
@@ -1057,18 +1057,27 @@ void PyRegion::createSpec(const char * nodeType, Spec & ns, const char* classNam
     NTA_ASSERT(input.getItem("required") != nullptr)
         << inputMessagePrefix.str() << "required";
     bool required = py::Int(input.getItem("required")) != 0;
-
-    NTA_ASSERT(input.getItem("regionLevel") != nullptr)
-        << inputMessagePrefix.str() << "regionLevel";
-    bool regionLevel = py::Int(input.getItem("regionLevel")) != 0;
+    
+    // make regionLevel optional and default to true.
+    bool regionLevel = true;
+    if (input.getItem("regionLevel") != nullptr)
+    {
+       regionLevel = py::Int(input.getItem("regionLevel")) != 0;
+    }
 
     NTA_ASSERT(input.getItem("isDefaultInput") != nullptr)
         << inputMessagePrefix.str() << "isDefaultInput";
     bool isDefaultInput = py::Int(input.getItem("isDefaultInput")) != 0;
-
-    NTA_ASSERT(input.getItem("requireSplitterMap") != nullptr)
-        << inputMessagePrefix.str() << "requireSplitterMap";
-    bool requireSplitterMap = py::Int(input.getItem("requireSplitterMap")) != 0;
+    
+    // make requireSplitterMap optional and default to false.
+    bool requireSplitterMap = false;
+    if (input.getItem("requireSplitterMap") != nullptr)
+    {
+      requireSplitterMap =  py::Int(input.getItem("requireSplitterMap")) != 0;
+    }
+    
+    // As of 6/20/16 regionLevel and require SplitterMap are optional parameters
```

Comment:
I don't think we need this comment - the date of the change is irrelevant. In this case no one is really using this parameter so this is a very minor change.